lib/knapsack_pro/extensions/rspec_extension.rb

```diff
@@ -0,0 +1,164 @@
+    # You don't break the Law of Demeter when you use a `knapsack__` method. You can use complex or chained internal RSpec methods that are hidden inside of a `knapsack__` method.
+    #
+    # Often, the structure of the `knapsack__` method is inspired by an existing RSpec method. The `knapsack__` method should have a similar structure to the original RSpec method.
+    # It should be easy to see a code difference when a new RSpec version is released.
+    # Please use permalinks to RSpec source code in your comments when you create a new `knapsack__` method.
+    module RSpecExtension
+      def self.setup!
+        RSpec::Core::World.prepend(World)
+        RSpec::Core::Runner.prepend(Runner)
+        RSpec::Core::Configuration.prepend(Configuration)
+      end
+
+      module World
+        # Based on `announce_filters`
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/world.rb#L171
+        def knapsack__setup
+          fail_if_config_and_cli_options_invalid
+        end
+      end
+
+      module Runner
+        def knapsack__mark_setup_as_done
+          @knapsack__setup_done = true
+        end
+
+        def knapsack__setup_done?
+          !!@knapsack__setup_done
+        end
+
+        # Based on:
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/runner.rb#L98
+        def knapsack__setup(stream_error = $stderr, stream_out = $stdout)
+          # Abstract from RSpec::Core::Runner#setup, since we do not need to set any filters or files at this point,
+          # and we do not want to let world.announce_filters be called, since it will print
+          # out `No examples found.` message.
+          configure(stream_error, stream_out)
+
+          world.knapsack__setup
+
+          knapsack__mark_setup_as_done
+        end
+
+        def knapsack__wants_to_quit?
+          world.wants_to_quit
+        end
+
+        # `rspec_is_quitting` was added in RSpec 3.11.0
+        # Changelog:
+        #   https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/Changelog.md#3110--2022-02-09
+        # PR:
+        #   https://github.com/rspec/rspec-core/pull/2926
+        # We support RSpec < 3.11.0.
+        def knapsack__rspec_is_quitting?
+          world.respond_to?(:rspec_is_quitting) && world.rspec_is_quitting
+        end
+
+        def knapsack__exit_early
+          _exit_status = configuration.reporter.exit_early(exit_code)
+        end
+
+        # Based on:
+        # https://github.com/rspec/rspec-core/blob/f8c8880dabd8f0544a6f91d8d4c857c1bd8df903/lib/rspec/core/configuration.rb#L546
+        # Set with --error-exit-code CODE - Override the exit code used when there are errors loading or running specs outside of examples.
+        def knapsack__error_exit_code
+          _default = nil || configuration.error_exit_code
```

Comment:
It would. I added in or purpose so that by looking at this code you know it's possible to get a nil and it's default.

Reply:
It's confusing to me because `nil || whatever` will always evaluate to `whatever`. I'd prefer a comment if you want to convey that:

```suggestion
          configuration.error_exit_code # nil unless `--failure-exit-code` is specified
```

Reply:
Agreed with Ryś.

In general, I like how you Artur are seemingly moving towards using code instead of comments to convey the reasoning behind the code. :+1: In this specific case however, I think the comment is just fine and actually easier to parse mentally.

Reply:
fix: https://github.com/KnapsackPro/knapsack_pro-ruby/pull/237/commits/278638a9dcf455b69b769a59f4b528d44a66d4a9